packages/tc-ui/src/primitives/number/server.jsx

```diff
@@ -4,6 +4,9 @@ module.exports = {
 	name: 'Number',
 	EditComponent: text.EditComponent,
 	ViewComponent: text.ViewComponent,
-	parser: value => (Number.isNaN(Number(value)) ? value : Number(value)),
+	parser: value => {
+		if (value === '') return null;
+		return Number.isNaN(Number(value)) ? value : Number(value);
+	},
 	hasValue: value => value || value === 0,
```

Comment:
does this line need to change too? A related bug, I guess the current implementation means the field shows up on the view page showing '0' even when it was originally empty.  Not 100% sure of that, but worth checking and fixing if necessary

Reply:
https://github.com/Financial-Times/treecreeper/blob/300c6a1214758ee53f0fb25d00e904e7c6f84019/packages/tc-ui/src/lib/components/structure.jsx#L67 is why I think it might cause this bad behaviour

Reply:
actually, I don't think this is a problem - ignore me